### PR TITLE
Adds Steam reforming and metallic hydrogen reactions

### PR DESCRIPTION
**Pull request by AtnerNT**
Adds the reaction to form metallic hydrogen using very high temperatures and pressures.

Adds Steam reforming, a process in which plasma is reacted with water vapor at temperatures of 200 degrees C or more to produce carbon monoxide and hydrogen gas.
This reaction is highly endothermic.
Carbon monoxide currently doesn't have any uses but functions as an invisible noxious gas.

### Changelog:

CL: [New] Adds carbon monoxide and its relevant canister
CL: [New] Adds metallic hydrogen reaction
CL: [New] Steam reforming reaction to produce hydrogen and carbon monoxide

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Reactions/HydrogenFormation.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	public class HydrogenFormation : Reaction
+	{
+		public bool Satisfies(GasMix gasMix)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public void React(GasMix gasMix, MetaDataNode node)
+		{
+			Debug.Log("Hydrogen is trying to form");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Reactions/HydrogenFormation.cs]:
```suggestion

```

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Reactions/MetalHydrogenFormation.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			{
+				var crystalliseRate = gasMix.GetMoles(Gas.Hydrogen) * temperatureScale / AtmosDefines.HYDROGEN_CRYSTALLISE_RATE;
+				
+				if (crystalliseRate > 0.0001f)
+				{
+					gasMix.RemoveGas(Gas.Hydrogen, crystalliseRate);
+
+					if (gasMix.Temperature > AtmosDefines.HYRDOGEN_MIN_CRYSTALLISE_TEMPERATURE && gasMix.Temperature < AtmosDefines.HYRDOGEN_MAX_CRYSTALLISE_TEMPERATURE && rnd.Next(0, 2) == 0)
+					{
+						SpawnSafeThread.SpawnPrefab(node.Position, AtmosManager.Instance.metalHydrogen);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Reactions/MetalHydrogenFormation.cs]: This needs to be world position

```node.Position.ToWorldInt(node.PositionalMatrix);``` Think that should be right

**Comment by AtnerNT** [comment]
ugh, bunch of reagent asset changes due to an index change, but that should fix tests

**Comment by AtnerNT** [comment]
Codacy is complaing, read through what is saying, and I don't think any of them are issues especially in context of the scripts they are in so I think its okay. I'm happy to make chances if believed nessacary but I think its codacy making a fuss for nothing.

**Comment by corp-0** [comment]
I do agree with all the issues codacy raised

**Comment by AtnerNT** [comment]
Aight codacy issues addressed, should pass all checks now

File: UnityProject/Assets/Scripts/ScriptableObjects/Atmospherics/GasesSingleton.cs [hunk 3]
```diff
@@ -20,24 +20,77 @@ public class GasesSingleton : SingletonScriptableObject<GasesSingleton>
...
-		public GasSO Nitryl = null;
-		public GasSO Tritium = null;
-		public GasSO HyperNoblium = null;
-		public GasSO Stimulum = null;
-		public GasSO Pluoxium = null;
-		public GasSO Freon = null;
-		public GasSO Smoke = null;
-		public GasSO Ash = null;
-		public GasSO CarbonMonoxide = null;
+		[SerializeField]
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/ScriptableObjects/Atmospherics/GasesSingleton.cs]: this one is optional and up to you but could you change
```cs
[SerializedField]
private GasSO plasma = null;
public GasSO Plasma => plasma;
```

to 

```cs
[field: SerializedField]
public GasSO Plasma { get; private set; }
```

**Comment by AtnerNT** [comment]
Changes applied, good to merge?

**Comment by corp-0** [comment]
codewise, yeah. Did it work in inspector? It should keep all references to gases

**Comment by AtnerNT** [comment]
It didnt keep the references but I reassinged them all in the same commit

**Comment by corp-0** [comment]
awesome

**Review by corp-0 — APPROVED** [review]

**Pull request merged**